src/common/Session.cpp

```diff
@@ -0,0 +1,158 @@
+/***************************************************************************
+* Copyright (c) 2015 Pier Luigi Fiorini <pierluigi.fiorini@gmail.com>
+*
+* This program is free software; you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation; either version 2 of the License, or
+* (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with this program; if not, write to the
+* Free Software Foundation, Inc.,
+* 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+***************************************************************************/
+
+#include <QFile>
+#include <QFileInfo>
+#include <QTextStream>
+
+#include "Configuration.h"
+#include "Session.h"
+
+const QString entryExtention = QStringLiteral(".desktop");
+
+namespace SDDM {
+    Session::Session()
+        : m_valid(false)
+    {
+    }
+
+    Session::Session(Type type, const QString &name)
+    {
+        setTo(type, name);
+    }
+
+    bool Session::isValid() const
+    {
+        return m_valid;
+    }
+
+    Session::Type Session::type() const
+    {
+        return m_type;
+    }
+
+    QString Session::xdgSessionType() const
+    {
+        return m_xdgSessionType;
+    }
+
+    QString Session::name() const
+    {
+        return m_name;
+    }
+
+    QDir Session::directory() const
+    {
+        return m_dir;
+    }
+
+    QString Session::fileName() const
+    {
+        return m_fileName;
+    }
+
+    QString Session::displayName() const
+    {
+        return m_displayName;
+    }
+
+    QString Session::comment() const
+    {
+        return m_comment;
+    }
+
+    QString Session::exec() const
+    {
+        return m_exec;
+    }
+
+    QString Session::tryExec() const
+    {
+        return m_tryExec;
+    }
+
+    QString Session::desktopNames() const
+    {
+        return m_desktopNames;
+    }
+
+    void Session::setTo(Type type, const QString &name)
```

Comment:
so the api is that the name argument should include ".desktop" , but ::name()  won't?

I'd call the arg fileName to be less confusing.